Scripts/Tracking/CollisionTracker.cs

```diff
@@ -0,0 +1,160 @@
+        public CollisionTrackerUnityEvent CollisionExit = new CollisionTrackerUnityEvent();
+        /// <summary>
+        /// Emitted when entering a <see cref="Collider"/> with <see cref="Collider.isTrigger"/> set.
+        /// </summary>
+        public CollisionTrackerUnityEvent TriggerEnter = new CollisionTrackerUnityEvent();
+        /// <summary>
+        /// Emitted as long as a collision with a <see cref="Collider"/> with <see cref="Collider.isTrigger"/> set happens.
+        /// </summary>
+        public CollisionTrackerUnityEvent TriggerStay = new CollisionTrackerUnityEvent();
+        /// <summary>
+        /// Emitted when exiting a <see cref="Collider"/> with <see cref="Collider.isTrigger"/> set.
+        /// </summary>
+        public CollisionTrackerUnityEvent TriggerExit = new CollisionTrackerUnityEvent();
+
+        protected virtual void OnCollisionEnter(Collision collision)
+        {
+            OnCollisionEnterEvent(
+                new CollisionTrackerData
+                {
+                    isTrigger = false,
+                    collision = collision,
+                    collider = collision.collider
+                });
+        }
+
+        protected virtual void OnCollisionStay(Collision collision)
+        {
+            OnCollisionStayEvent(
+                new CollisionTrackerData
+                {
+                    isTrigger = false,
+                    collision = collision,
+                    collider = collision.collider
+                });
+        }
+
+        protected virtual void OnCollisionExit(Collision collision)
+        {
+            OnCollisionExitEvent(
+                new CollisionTrackerData
+                {
+                    isTrigger = false,
+                    collision = collision,
+                    collider = collision.collider
+                });
+        }
+
+        protected virtual void OnTriggerEnter(Collider collider)
+        {
+            OnTriggerEnterEvent(
+                new CollisionTrackerData
+                {
+                    isTrigger = true,
+                    collision = null,
+                    collider = collider
+                });
+        }
+
+        protected virtual void OnTriggerStay(Collider collider)
+        {
+            OnTriggerStayEvent(
+                new CollisionTrackerData
+                {
+                    isTrigger = true,
+                    collision = null,
+                    collider = collider
+                });
+        }
+
+        protected virtual void OnTriggerExit(Collider collider)
+        {
+            OnTriggerExitEvent(
+                new CollisionTrackerData
+                {
+                    isTrigger = true,
+                    collision = null,
+                    collider = collider
+                });
+        }
+
+        protected void OnCollisionEnterEvent(CollisionTrackerData data)
+        {
+            CollisionEnter?.Invoke(data, this);
```

Comment:
wrap these with `isActiveAndEnabled`

I'm wondering if the `OnTriggerEnter` and the others still work if the component is disabled?

If not then don't bother.

Reply:
Turns out Unity still calls those "message methods". Added the check in!